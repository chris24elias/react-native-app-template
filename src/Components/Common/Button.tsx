import React from 'react';
import {
  Button as KittenButton,
  ButtonProps as KittenButtonProps,
  Text,
} from '@ui-kitten/components';
import {TextStyle} from 'react-native';
import useTheme from '../../Theme';

interface ButtonProps extends KittenButtonProps {
  text: string;
  textStyle?: TextStyle;
}

const Button = ({
  text,
  textStyle,
  style,
  children,
  ...otherProps
}: ButtonProps) => {
  const {theme} = useTheme();

  return (
    <KittenButton
      style={[
        {backgroundColor: theme.Primary.main, borderColor: theme.Primary.main},
        style,
      ]}
      activeOpacity={0.2}
      {...otherProps}>
      {(evaProps) => (
        <Text {...evaProps} style={[evaProps.style, textStyle]}>
          {text}
        </Text>
      )}
    </KittenButton>
  );
};

export default Button;
