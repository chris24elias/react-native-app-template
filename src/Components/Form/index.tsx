import * as React from 'react';
import {TextInput, ViewStyle, ScrollViewProps} from 'react-native';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';

interface Props {
  children: JSX.Element | JSX.Element[];
  register;
  setValue;
  errors;

  //   ------------------------
  scrollview?: boolean;
  showsVerticalScrollIndicator?: boolean;
  contentContainerStyle?: ViewStyle;
  scrollViewProps?: ScrollViewProps;
}

export default ({
  contentContainerStyle,
  scrollview,
  showsVerticalScrollIndicator,
  scrollViewProps,
  children,
  register,
  setValue,
  errors,
}: Props) => {
  const InputRefs = React.useRef<any[]>([]);
  const InputNames = React.useRef<string[]>([]);

  React.useEffect(() => {
    InputNames?.current?.forEach((name) => {
      register({name});
    });
  }, [register]);

  const parseChildren = (
    children: any,
    counter: {count: number} = {count: 0},
  ): any => {
    return (Array.isArray(children) ? [...children] : [children]).map(
      (child) => {
        if (child?.props?.children?.type && child?.props?.name) {
          return parseChildren(child?.props?.children, counter);
        }
        console.log('child', child?.props);
        return child?.props?.name ? parseChild(counter, child) : child;
      },
    );
  };

  const focusNextTextInput = (index) => {
    // find next textinput ref and focus it

    for (let i = index + 1; i < InputRefs.current.length; i++) {
      const input = InputRefs.current[i];
      if (input?.focus) {
        input.focus();
        return;
      }
    }

    // if no textinputs found, just blur

    InputRefs?.current[index]?.blur();
  };

  const parseChild = (counter: {count: number}, child: any) => {
    let index = counter.count;
    counter.count++;
    return React.createElement(child.type, {
      ...{
        ...child.props,
        ref: (e: TextInput) => {
          // register({name: child.props.name});
          InputRefs.current[index] = e;
          InputNames.current[index] = child.props.name;
        },
        onChangeText: (v: string) => setValue(child.props.name, v, true),
        onSubmitEditing: () => {
          focusNextTextInput(index);
        },
        // blurOnSubmit: false,
        key: child.props.name,
        error: errors[child.props.name],
      },
    });
  };

  if (scrollview) {
    return (
      <KeyboardAwareScrollView
        showsVerticalScrollIndicator={showsVerticalScrollIndicator}
        contentContainerStyle={contentContainerStyle}
        enableResetScrollToCoords={false}
        extraScrollHeight={50}
        extraHeight={100}
        keyboardShouldPersistTaps={'handled'}
        {...scrollViewProps}>
        {parseChildren(children)}
      </KeyboardAwareScrollView>
    );
  }

  return <>{parseChildren(children)}</>;
};
