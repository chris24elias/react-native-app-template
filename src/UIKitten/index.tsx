import React from 'react';
import * as eva from '@eva-design/eva';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';
import {EvaIconsPack} from '@ui-kitten/eva-icons';

interface Props {
  children: any;
}

const UIKitten = ({children}: Props) => {
  return (
    <ApplicationProvider {...eva} theme={eva.light}>
      <IconRegistry icons={EvaIconsPack} />
      {children}
    </ApplicationProvider>
  );
};

export default UIKitten;
