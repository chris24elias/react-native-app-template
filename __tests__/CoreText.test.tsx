/**
 * @jest-environment jsdom
 */

import React from 'react';
import {shallow} from 'enzyme';
import {Text} from '../src/Components/Common';
import UIKitten from '../src/UIKitten';
import ThemeContextProvider from '../src/Theme/ThemeContext';

describe('Common Text', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <ThemeContextProvider>
          <UIKitten>
            <Text text="test label" />
          </UIKitten>
        </ThemeContextProvider>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
