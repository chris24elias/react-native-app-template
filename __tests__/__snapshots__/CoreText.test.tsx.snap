// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Text Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
